modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/security/permissions/resource/FDSViewsPermission.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.data.set.views.web.internal.security.permissions.resource;
+
+import com.liferay.frontend.data.set.views.web.internal.constants.FDSViewsConstants;
+import com.liferay.osgi.util.service.Snapshot;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+
+/**
+ * @author Kevin Tan
+ */
+public class FDSViewsPermission {
+
+	public static boolean contains(
+		PermissionChecker permissionChecker, long groupId, String actionId) {
+
+		PortletResourcePermission portletResourcePermission =
+			_portletResourcePermissionSnapshot.get();
+
+		return portletResourcePermission.contains(
+			permissionChecker, groupId, actionId);
+	}
+
+	private static final Snapshot<PortletResourcePermission>
+		_portletResourcePermissionSnapshot = new Snapshot<>(
+			FDSViewsPermission.class, PortletResourcePermission.class,
+			"(resource.name=" + FDSViewsConstants.RESOURCE_NAME + ")");
```

Comment:
What will happen when we want to add permissions for dataset views? These permissions are really for datasets (FDSEntries). I feel that the View vs Entry naming is inconsistent for this class, resource name, and multiple other places in the PR. 

Zooming out, can multiple sets of permissions even be defined in a single module? I feel that we might need to split this module into several, something like:
- data-set
  - data-set-entries-web // with permissions for DS entries
  - data-set-views-web // with permissions for DS views

Sidenote: additional modules may be
- data-set
  - data-set-web // common JS components and utils
  - data-set-item-selector-web // fragment item selector

Reply:
Dataset views are object entries so they should already have permissions integrated according to https://learn.liferay.com/w/dxp/building-applications/objects/understanding-object-integrations/permissions-framework-integration.

The `FDSViews*` naming is based on the module name, so it seemed the most consistent to keep it with that. I don't think we'd need other names since the only unique permission to be added seems to be a `MANAGE_FDS_VIEWS_ENTRIES` key.

I was also considering using permissions directly from the `FDS Entry` object, but I'm not sure how to access that from a util class or how an end user would manage that unless they'd go directly to the objects manager.

Reply:
I am not sure either. That was roughly the point of the spike:
- How to double same UI that we have in Objects? Can we do something ugly like import a JSP from objects-web? Is it even possible to manage permissions from a module that does not strictly define the entity (ObjectEntry)?
- How to add the additional permission to create views? Can we append to existing permissions? Should Objects team maybe generate that permission when a relationship is created?

Reply:
> I was also considering using permissions directly from the FDS Entry object, but I'm not sure how to access that from a util class or how an end user would manage that unless they'd go directly to the objects manager.

I think this worths a try for the resources themselves.

Does the `FDSViewsPermission` class provide checker for the portlet-related actions only?